src/payment/strategies/clearpay/clearpay-payment-strategy.ts

```diff
@@ -0,0 +1,113 @@
+import { CheckoutStore, CheckoutValidator, InternalCheckoutSelectors } from '../../../checkout';
+import { InvalidArgumentError, MissingDataError, MissingDataErrorType, NotInitializedError, NotInitializedErrorType } from '../../../common/error/errors';
+import { OrderActionCreator, OrderRequestBody } from '../../../order';
+import { StoreCreditActionCreator } from '../../../store-credit';
+import { PaymentArgumentInvalidError } from '../../errors';
+import PaymentActionCreator from '../../payment-action-creator';
+import PaymentMethod from '../../payment-method';
+import PaymentMethodActionCreator from '../../payment-method-action-creator';
+import { PaymentInitializeOptions, PaymentRequestOptions } from '../../payment-request-options';
+import PaymentStrategy from '../payment-strategy';
+
+import ClearpayScriptLoader from './clearpay-script-loader';
+import ClearpaySdk from './clearpay-sdk';
+
+export default class ClearpayPaymentStrategy implements PaymentStrategy {
+    private _clearpaySdk?: ClearpaySdk;
+
+    constructor(
+        private _store: CheckoutStore,
+        private _checkoutValidator: CheckoutValidator,
+        private _orderActionCreator: OrderActionCreator,
+        private _paymentActionCreator: PaymentActionCreator,
+        private _paymentMethodActionCreator: PaymentMethodActionCreator,
+        private _storeCreditActionCreator: StoreCreditActionCreator,
+        private _clearpayScriptLoader: ClearpayScriptLoader
+    ) {}
+
+    async initialize(options: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
+        const { paymentMethods: { getPaymentMethodOrThrow } } = this._store.getState();
+        const paymentMethod = getPaymentMethodOrThrow(options.methodId, options.gatewayId);
+
+        this._clearpaySdk = await this._clearpayScriptLoader.load(paymentMethod);
+
+        return this._store.getState();
+    }
+
+    deinitialize(): Promise<InternalCheckoutSelectors> {
+        this._clearpaySdk = undefined;
+
+        return Promise.resolve(this._store.getState());
+    }
+
+    async execute(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        if (!payload.payment) {
+            throw new PaymentArgumentInvalidError(['payment.gatewayId', 'payment.methodId']);
+        }
+        const { gatewayId, methodId } = payload.payment;
+
+        if (!gatewayId || !methodId) {
+            throw new PaymentArgumentInvalidError(['payment.gatewayId', 'payment.methodId']);
+        }
+
+        const { isStoreCreditApplied: useStoreCredit } = this._store.getState().checkout.getCheckoutOrThrow();
+        let state = this._store.getState();
+
+        if (useStoreCredit !== undefined) {
+            state = await this._store.dispatch(this._storeCreditActionCreator.applyStoreCredit(useStoreCredit));
+        }
+
+         // Validate nothing has changed before redirecting to Clearpay checkout page
```

Comment:
rather than having this as a comment, you can remove it since you already have a unit test for it. You could rename the unit test from
```it('validates the checkout')```
to
```it('validates nothing has changed before redirecting to Clearpay checkout page`)```